contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpPaginatorConfig.java

```diff
@@ -69,6 +60,85 @@
   @JsonProperty
   private final String method;
 
+  public HttpPaginatorConfig(HttpPaginatorConfigBuilder builder) {
+    this.limitParam = builder.limitParam;
+    this.offsetParam = builder.offsetParam;
+    this.pageParam = builder.pageParam;
+    this.pageSizeParam = builder.pageSizeParam;
+    this.pageSize = builder.pageSize;
+    this.maxRecords = builder.maxRecords;
+    this.method = builder.method;
+  }
+
+  public static HttpPaginatorConfigBuilder builder() {
+    return new HttpPaginatorConfigBuilder();
+  }
+
+  public String limitParam() {
+    return this.limitParam;
+  }
+
+  public String offsetParam() {
+    return this.offsetParam;
+  }
+
+  public String pageParam() {
+    return this.pageParam;
+  }
+
+  public String pageSizeParam() {
+    return this.pageSizeParam;
+  }
+
+  public int pageSize() {
+    return this.pageSize;
+  }
+
+  public int maxRecords() {
+    return this.maxRecords;
+  }
+
+  public String method() {
+    return this.method;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    HttpPaginatorConfig that = (HttpPaginatorConfig) o;
+    return pageSize == that.pageSize
+      && maxRecords == that.maxRecords
+      && Objects.equals(limitParam, that.limitParam)
+      && Objects.equals(offsetParam, that.offsetParam)
+      && Objects.equals(pageParam, that.pageParam)
+      && Objects.equals(pageSizeParam, that.pageSizeParam)
+      && Objects.equals(method, that.method);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(limitParam, offsetParam, pageParam, pageSizeParam,
+      pageSize, maxRecords, method);
+  }
+
+  @Override
+  public String toString() {
```

Comment:
There is a `PlanStringBuilder` that can help with this tedium.

Reply:
@paul-rogers Thanks for the review, this pull request was created by Vova. Anyway, it looks like we'll be able to merge the request soon..